cip-190 deprecations (#202)
neo4j/docs-status-codes: modules/ROOT/pages/notifications/all-notifications.adoc

======
=====

> .Using an unescaped variable named 'where' in node

Author: ```suggestion
.Using an unescaped variable named 'where' in a node pattern
```

[.tabbed-example]
=====
[.include-with-neo4j-code]
======

Query::
+
[source,cypher]
----
MATCH (where {p: 5})
RETURN where
----

Description of the returned code::
`(where {p: 5})` is deprecated. It is replaced by `(++`where`++ {p: 5})`.

Suggestions for improvement::
To continue using variables with this name, use backticks to escape the variable name:
+
[source,cypher]
----
MATCH (`where` {p: 5})
RETURN `where`.p
----
======

[.include-with-GQLSTATUS-deprecated-with-replacement]
======
Query::
+
[source,cypher]
----
MATCH (where {p: 5})
RETURN where
----

Returned GQLSTATUS code::
01N01

Returned status description::
warn: feature deprecated with replacement.
`(where {p: 5})` is deprecated.
It is replaced by `(++`where`++ {p: 5})`.

Suggestions for improvement::
To continue using variables with this name, use backticks to escape the variable name:
+
[source,cypher]
----
MATCH (`where` {p: 5})
RETURN `where`.p
----
======
=====

> .Using an unescaped variable named 'where' in relationship patterns

Author: ```suggestion
.Using an unescaped variable named 'where' in a relationship pattern
```
For consistency with the above suggestion. Plural vs. singular is your call. It just seems to me singular is used more on this page.

`WHEN is :: INTEGER` is deprecated. It is replaced by `WHEN ++`is`++ :: INTEGER`.

Suggestions for improvement::
To continue using variables with this name in simple `CASE` expressions, use backticks to escape the variable name:
+
[source,cypher]
----
MATCH (n)
WITH n, n.internationalStandard AS `is`
RETURN CASE n
  WHEN `is` :: INTEGER THEN "ISO/IEC" + `is`
  ELSE `is`
END AS standardsName
----
======
[.include-with-GQLSTATUS-deprecated-with-replacement]
======
Query::
+
[source,cypher]
----
MATCH (n)
WITH n, n.internationalStandard AS is
RETURN CASE n
  WHEN is :: INTEGER THEN "ISO/IEC" + is
  ELSE is
END AS standardsName
----

Returned GQLSTATUS code::
01N01

Returned status description::
warn: feature deprecated with replacement.
`WHEN is pass:[::] INTEGER` is deprecated.
It is replaced by `WHEN ++`is`++ pass:[::] INTEGER`.

Suggestions for improvement::
To continue using variables with this name in simple `CASE` expressions, use backticks to escape the variable name:
+
[source,cypher]
----
MATCH (n)
WITH n, n.internationalStandard AS `is`
RETURN CASE n
  WHEN `is` :: INTEGER THEN "ISO/IEC" + `is`
  ELSE `is`
END AS standardsName
----
======
=====

> .Using an unescaped variable named `contains` for relationships in addition or subtraction operations within a `WHEN` operand in a simple `CASE` expression

Author: ```suggestion
.Using an unescaped variable named `contains` in addition or subtraction operations within a `WHEN` operand in a simple `CASE` expression
```

The variable contains in the example bind to an INTEGER (the number of relationships in the path) and that is primarily a characteristics of the specific examples.

As more abstract example would be:
```
WITH 2 AS x, 1 AS contains
RETURN CASE x
  WHEN contains + 1 THEN "foo"
  ELSE "bar"
END AS check
```
This does not involve any nodes and relationships — to make the point that nodes and relationships are not integral to this deprecations.

======
=====

> .Using an unescaped variable named `contains` for relationships in addition or subtraction operations within a `WHEN` operand in a simple `CASE` expression

Author: The variable `contains` in the example bind to an INTEGER (the number of relationships in the path) and that is primarily a characteristics of the specific examples. 

As more abstract example would be:
```
WITH 2 AS x, 1 AS contains
RETURN CASE x
  WHEN contains + 1 THEN "foo"
  ELSE "bar"
END AS check
```
This does not involve any nodes and relationships — to make the point that nodes and relationships are not integral to this deprecations.
